lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -207,6 +210,51 @@ void SPIRVRegularizeLLVM::lowerFunnelShiftLeft(IntrinsicInst *FSHLIntrinsic) {
   FSHLIntrinsic->setCalledFunction(FSHLFunc);
 }
 
+void SPIRVRegularizeLLVM::buildUMulWithOverflowFunc(Function *UMulFunc) {
+  if (!UMulFunc->empty())
+    return;
+
+  BasicBlock *EntryBB = BasicBlock::Create(M->getContext(), "entry", UMulFunc);
+  IRBuilder<> Builder(EntryBB);
+  // Build the actual unsigned multiplication logic with the overflow
+  // indication.
+  auto *FirstArg = UMulFunc->getArg(0);
+  auto *SecondArg = UMulFunc->getArg(1);
+
+  // Do unsigned multiplication Mul = A * B.
+  // Then check if unsigned division Div = Mul / A is not equal to B.
+  // If so, then overflow has happened.
+  auto *Mul = Builder.CreateNUWMul(FirstArg, SecondArg);
+  auto *Div = Builder.CreateUDiv(Mul, FirstArg);
+  auto *Cmp = Builder.CreateICmpNE(FirstArg, Div);
+
+  Type *BoolTy = UMulFunc->getReturnType()->getContainedType(1);
+  auto *TrueVal = Constant::getAllOnesValue(BoolTy);
+  auto *FalseVal = Constant::getNullValue(BoolTy);
```

Comment:
Probably `ConstantInt::getTrue` and `ConstantInt::getFalse` would be better from readability point here